From 694fe636bb23b9fe02e5ff3143ae46607c6a46af Mon Sep 17 00:00:00 2001
From: AKASHI Takahiro <takahiro.akashi@linaro.org>
Date: Fri, 30 May 2014 11:23:41 +0900
Subject: [PATCH] arm64: ftrace: (bugfix) synced with ftcace interface change

ftrace_init() failed since ftrace_dyn_arch_init() doesn't initialize
the argument to null.
This bug comes in only if arm64 ftrace is back-ported as
ftrace_dyn_arch_init() interface has been changed in the merge window of
3.15.

Signed-off-by: AKASHI Takahiro <takahiro.akashi@linaro.org>
Signed-off-by: Mark Brown <broonie@linaro.org>
(cherry picked from LSK commit 0cc5286fc3ca12ec0d388e4d8c08a66b40d52233)
Signed-off-by: Alex Shi <alex.shi@linaro.org>
---
 arch/arm64/kernel/ftrace.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/arch/arm64/kernel/ftrace.c b/arch/arm64/kernel/ftrace.c
index 2e61b95..649890a 100644
--- a/arch/arm64/kernel/ftrace.c
+++ b/arch/arm64/kernel/ftrace.c
@@ -94,6 +94,7 @@ int ftrace_make_nop(struct module *mod, struct dyn_ftrace *rec,
 
 int __init ftrace_dyn_arch_init(void *data)
 {
+	*(unsigned long *)data = 0;
 	return 0;
 }
 #endif /* CONFIG_DYNAMIC_FTRACE */
-- 
1.8.0

